test/test/disconnected.cpp

```diff
@@ -105,6 +105,8 @@ TEST_CASE("disconnected,handler,3")
         });
 
     WaitForSingleObject(signal.get(), INFINITE);
+    // Give some time for to_hresult() to complete.
+    Sleep(500);
```

Comment:
Shouldn't we build this buffer into the test that fiddles with `winrt_to_hresult_handler` rather than the test(s) that observe the fiddling? I think there's only one place where this is tested (`custom_error.cpp`).

Reply:
I think putting this in the disconnected test itself is better because it makes sure (or at least tries to) that any code related to this test running on the worker thread has really finished before leaving the test scope. In case it actually causes `std::terminate`, Catch2 can report that the abnormal termination happens during the correct test, instead of making it look like another test which happens to be running afterwards caused it. (This actually did happen to these tests before PR #1225.)